gherkin/php/tests/Parser/RuleTypeTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cucumber\Gherkin\Parser;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+
+final class RuleTypeTest extends TestCase
+{
+    /** @dataProvider tokenCaseProvider */
+    public function testItCanCastFromTokens(TokenType $tokenType): void
+    {
+        self::assertInstanceOf(RuleType::class, RuleType::cast($tokenType));
+    }
+
+    /**
+     * @return Generator<string,array{0:TokenType}>
+     */
+    public function tokenCaseProvider(): Generator
```

Comment:
I also tend to go with `iterable` rather than `Generator` as it lets you switch to `array` without further changes

Reply:
Nothing depends on this, so switching to `array` later is trivial, no?